src/controllers/ShowController.js

```diff
@@ -129,18 +120,18 @@ export default class ShowController {
       }
 
       return Show.aggregate([{
-          $sort: sort
-        }, {
           $match: query
```

Comment:
To use $text in the $match stage, the $match stage has to be the first stage of the pipeline.
Views do not support text search.

https://docs.mongodb.com/manual/reference/operator/aggregation/match/#restrictions